### PR TITLE
[RNE Rewrite] feat(models): wire the published Core ML and MLX builds

### PR DESCRIPTION
**Pull request by msluszniak**
## Description

Wires the Core ML and MLX builds published for v0.10.0 into `models.ts`, and drops entries whose artifact no longer exists at that tag.

### Introduces a breaking change?

- [x] Yes
- [ ] No

### Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Documentation update (improves or adds clarity to existing documentation)
- [ ] Other (chores, tests, code style improvements etc.)

### Tested on

- [ ] iOS
- [ ] Android

### Testing instructions

Every `.pte` URL in `models.ts` resolves against the Hub (150/150 return 200), `yarn typecheck` and `prettier --check` pass. The wiring is not device-tested, so each entry below needs a load-and-run check.

**Added, Core ML fp16** (iOS only):

| model | accessor |
| --- | --- |
| CLIP image | `imageEmbeddings.CLIP_VIT_BASE_PATCH32.COREML_FP16` |
| DeepLabV3 resnet50 | `semanticSegmentation.DEEPLAB_V3_RESNET50.COREML_FP16` |
| DeepLabV3 resnet101 | `semanticSegmentation.DEEPLAB_V3_RESNET101.COREML_FP16` |
| DeepLabV3 mobilenet | `semanticSegmentation.DEEPLAB_V3_MOBILENET_V3_LARGE.COREML_FP16` |
| FCN resnet50 | `semanticSegmentation.FCN_RESNET50.COREML_FP16` |
| FCN resnet101 | `semanticSegmentation.FCN_RESNET101.COREML_FP16` |
| LR-ASPP | `semanticSegmentation.LRASPP_MOBILENET_V3_LARGE.COREML_FP16` |
| selfie portrait | `semanticSegmentation.SELFIE_SEGMENTATION.COREML_FP16` |
| selfie landscape | `semanticSegmentation.SELFIE_SEGMENTATION_LANDSCAPE.COREML_FP16` |
| RF-DETR detector | `objectDetection.RFDETR_NANO.COREML_FP16` |
| RF-DETR segmentation | `instanceSegmentation.RFDETR_NANO.COREML_FP16` |
| YOLO26, 5 scales x 3 sizes | `objectDetection.YOLO26.{NANO,SMALL,MEDIUM,LARGE,XLARGE}.SIZE_{384,512,640}.COREML_FP16` |

**Added, MLX int8** (iOS only):

| model | accessor |
| --- | --- |
| CLIP image | `imageEmbeddings.CLIP_VIT_BASE_PATCH32.MLX_INT8` |
| distiluse | `textEmbeddings.DISTILUSE_BASE_MULTILINGUAL_CASED_V2.MLX_INT8` |
| whisper tiny / tiny.en | `speechToText.WHISPER.TINY.MLX_INT8` / `speechToText.WHISPER.EN.TINY.MLX_INT8` |
| whisper base / base.en | `speechToText.WHISPER.BASE.MLX_INT8` / `speechToText.WHISPER.EN.BASE.MLX_INT8` |
| whisper small / small.en | `speechToText.WHISPER.SMALL.MLX_INT8` / `speechToText.WHISPER.EN.SMALL.MLX_INT8` |

**Added, XNNPACK fp32** (worth checking on Android, the artifact is newly published):

| model | accessor |
| --- | --- |
| selfie landscape | `semanticSegmentation.SELFIE_SEGMENTATION_LANDSCAPE.XNNPACK_FP32` |

**Not wired: distiluse Core ML.** It was in an earlier revision of this PR and has been taken back out. The published artifact still fails [#1164](https://github.com/software-mansion/react-native-executorch/issues/1164): the exporter declares `Dim("tokens", min=1, max=126)` but the compiled Core ML model enforces `2..126`, because `torch.export` specializes size 1 away, so single-token input fails and anything longer works. Verified against the published v0.10.0 artifact on host. This affects every published Core ML text embedder (`all_minilm_l6_v2`, `clip_vit_base_patch32_text`, `multi_qa_minilm_l6_cos_v1`, `paraphrase_multilingual_minilm_l12_v2`), which is why none of them is wired. #1164 says to re-add once the export is fixed. distiluse `MLX_INT8` is unaffected and stays.

**Removed** (breaking; confirm nothing in the examples still references these):

| entry | why |
| --- | --- |
| `styleTransfer.{CANDY,MOSAIC,RAIN_PRINCESS,UDNIE}.COREML_FP32` | fp32 dropped at v0.10.0; the fp16 build it leaves behind is now faster than fp32 ever was, see below |
| `objectDetection.SSDLITE320_MOBILENET_V3_LARGE.COREML_FP32` | same |
| `instanceSegmentation.FASTSAM.{S,X}.COREML_FP32` | same |
| `objectDetection.RFDETR_NANO.COREML_INT8` and `instanceSegmentation.RFDETR_NANO.COREML_INT8` | replaced by the fp16 build, ~4x faster |
| `speechToText.WHISPER.SMALL.MLX_BF16` and `.EN.SMALL.MLX_BF16` | needs ~3.8 GB RSS, jetsam-killed on an iPhone 16 |
| `imageEmbeddings.CLIP_VIT_BASE_PATCH32.XNNPACK_INT8` | least accurate variant measured, cosine 0.879 |

**Republished since this PR opened.** Same URLs, same file sizes, faster artifacts, so no wiring change was needed:

| model | before | after | |
| --- | --- | --- | --- |
| style transfer, all four | 209.4 ms | 114.2 ms | 1.8x, pinned `compute_unit=CPU_AND_GPU` |
| `fast_sam` s | 52.5 ms | 8.7 ms | 6.0x, pinned `compute_unit=CPU_AND_NE` |
| `fast_sam` x | 108.9 ms | 38.9 ms | 2.8x, pinned `compute_unit=CPU_AND_NE` |

Core ML exports default to `ComputeUnit.ALL`, which chooses placement at runtime and chose badly for these two. This also settles the review question about dropping fp32: fp32 only measured faster than fp16 because fp32 is ineligible for the ANE, so the comparison was measuring placement, not precision. Pinned correctly, fp16 is faster than fp32 at half the size. Measured on an iPhone 16, ET 1.4.1, medians over interleaved repetitions; full four-way panel across eleven models in export-scripts MR 15.

**Retagged** v0.9.0 to v0.10.0: `efficientnet`, `ssdlite`, the four style transfer models, `selfie` and `lraspp`. Every file that exists at both tags is byte-identical except the four style transfer `coreml_fp16` builds, which were rebuilt with a pinned compute unit and are 1.8x faster (above). Otherwise the bump only picks up the fp32 removal on `efficientnet` and `ssdlite`, and the newly published Core ML builds on `selfie` and `lraspp`.

### Related issues

Publishing side: export-scripts MR 15.

### Checklist

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly
- [x] My changes generate no new warnings

### CODE REVIEW

File: packages/react-native-executorch/src/models.ts [hunk 1]
```diff
@@ -75,75 +75,59 @@ const STYLE_TRANSFER_OPTS = {
...
 const STYLE_TRANSFER_CANDY_XNNPACK_INT8: StyleTransferModel = {
-  modelPath: `${BASE_URL}-style-transfer-candy/${VERSION_TAG}/xnnpack/style_transfer_candy_xnnpack_int8.pte`,
+  modelPath: `${BASE_URL}-style-transfer-candy/${NEXT_VERSION_TAG}/xnnpack/style_transfer_candy_xnnpack_int8.pte`,
   modelOpts: STYLE_TRANSFER_OPTS,
 };
 const STYLE_TRANSFER_CANDY_COREML_FP16: StyleTransferModel = {
-  modelPath: `${BASE_URL}-style-transfer-candy/${VERSION_TAG}/coreml/style_transfer_candy_coreml_fp16.pte`,
-  modelOpts: STYLE_TRANSFER_OPTS,
-};
-const STYLE_TRANSFER_CANDY_COREML_FP32: StyleTransferModel = {
```
Comment (barhanc) [inline packages/react-native-executorch/src/models.ts:90]: From my tests on iPhone 17, the fp32 variant is 2x faster (~160ms vs ~300ms on fp16).

File: packages/react-native-executorch/src/models.ts [hunk 2]
```diff
@@ -1353,6 +1515,7 @@ export const models = {
           XNNPACK_FP32: WHISPER_BASE_EN_XNNPACK_FP32,
           COREML_FP16: WHISPER_BASE_EN_COREML_FP16,
           MLX_BF16: WHISPER_BASE_EN_MLX_BF16,
+          MLX_INT8: WHISPER_BASE_EN_MLX_INT8,
```
Comment (barhanc) [inline packages/react-native-executorch/src/models.ts:1792]: Here we are keeping both MLX precisions but in other Whisper variants the int8 replaces bf16.

File: packages/react-native-executorch/src/models.ts [hunk 3]
```diff
@@ -563,6 +632,14 @@ const DISTILUSE_BASE_MULTILINGUAL_CASED_V2_EMBEDDINGS: TextEmbedderModel = {
   modelPath: `${BASE_URL}-distiluse-base-multilingual-cased-v2/${NEXT_VERSION_TAG}/xnnpack/distiluse_base_multilingual_cased_v2_xnnpack_8da4w.pte`,
   tokenizerPath: `${BASE_URL}-distiluse-base-multilingual-cased-v2/${NEXT_VERSION_TAG}/tokenizer.json`,
 };
+const DISTILUSE_BASE_MULTILINGUAL_CASED_V2_COREML_FP16: TextEmbedderModel = {
```
Comment (barhanc) [inline packages/react-native-executorch/src/models.ts]: Does that resolve #1164 ?
Edit: Just tested and still the same issues

File: packages/react-native-executorch/src/models.ts (hunk 2 above)
Comment (msluszniak) [reply packages/react-native-executorch/src/models.ts:1792]: It is my deliberate decision. bf16 is a bit faster but at the same time 2 times bigger. bf16 on variant `small` crashes, so bf16 is not an option here.

File: packages/react-native-executorch/src/models.ts (hunk 3 above)
Comment (msluszniak) [reply packages/react-native-executorch/src/models.ts]: Yeah, the problem is this there unfortunately, I'll remove this model and document this in export scripts and/or upstream ET issue.

File: packages/react-native-executorch/src/models.ts (hunk 1 above)
Comment (msluszniak) [reply packages/react-native-executorch/src/models.ts:90]: I'm on it.

Comment (msluszniak) [reply packages/react-native-executorch/src/models.ts:90]: Ok, I tested and it is slower, because I incorrectly exported it with compute unit `ALL`, the best one is cpu only but I will stick to gpu and cpu since it is stable across runs and thermal throttling:
image
I will re-upload the f16 variant and ping you to re-test it.

Comment (msluszniak) [reply packages/react-native-executorch/src/models.ts:90]: @barhanc ok, style transfer is ready for re-testing.

**Review by barhanc — APPROVED** [review]

**Comment by msluszniak** [comment]
I just resolve this coreml distiluse and will merge.
